infra/Pos/DHT/Real/Real.hs

```diff
@@ -133,33 +147,26 @@ withKademliaLogger
     -> m a
 withKademliaLogger action = modifyLoggerName (<> "kademlia") action
 
+-- | Return a list of known peers.
+--
 -- You can get DHTNode using @toDHTNode@ and Kademlia function @peersToNodeIds@.
 kademliaGetKnownPeers
-    :: ( MonadIO m
-       , Mockable Async m
-       , Mockable Catch m
-       , Mockable Throw m
-       , Eq (Promise m (Maybe ()))
-       , WithLogger m
-       , Bi DHTData
-       , Bi DHTKey
-       )
-    => KademliaDHTInstance
-    -> m [NetworkAddress]
+    :: KademliaDHTInstance
+    -> STM [NetworkAddress]
```

Comment:
Moved this to `STM` so that the subscription worker can use it in a transaction. It makes that whole concurrency dance a lot easier to manage.